modules/conversantAnalyticsAdapter.js

```diff
@@ -377,19 +439,41 @@ cnvrHelper.createAdSize = function(width, height) {
 cnvrHelper.createAdUnit = function() {
   return {
     sizes: [],
+    mediaTypes: [],
     bids: {}
   };
 };
 
 /**
  * Helper to create a basic bid payload object.  By pre-creating the eventCodes we can easily push in our statuses.
-*/
+ */
 cnvrHelper.initializeBidDefaults = function() {
   return {
     'eventCodes': []
   };
 };
 
+
+/**
+ * Helper to get the sampling rates from an object and validate the result.
+ * @param parentObj Parent object that has the sampling property
+ * @param propNm Name of the sampling property
+ * @param defaultSampleRate A default value to apply if there is a problem
+ * @returns {number} returns a float number from 0 (always off) to 1 (always on)
+ */
+cnvrHelper.getSampleRate = function(parentObj, propNm, defaultSampleRate){
+  let sampleRate = defaultSampleRate;
+  if (parentObj && typeof parentObj === 'object' && parentObj.hasOwnProperty(propNm) && parentObj[propNm] !== 'undefined'){
```

Comment:
It's sufficient to check the following:
```javascript
if (parentObj && propNm && typeof parentObj[propNm] !== 'undefined')
```

Reply:
much cleaner thanks, javascript type checking is awful :(